tools/cache.go

```diff
@@ -0,0 +1,30 @@
+package tools
+
+import (
+	"github.com/dgraph-io/ristretto"
+	"time"
+)
+
+var cache *ristretto.Cache
+
+func initCache() {
+	c, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: 1e6,     // number of keys to track frequency of (1M).
+		MaxCost:     1 << 29, // 1 << 30 == 1/2GB Cost of cache
```

Comment:
Shouldn't this refer to `MaxCost`?
```suggestion
		MaxCost:     1 << 29, // 1 << 29 == 1/2GB Cost of cache
```